crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -811,12 +818,36 @@ impl SemanticSyntaxContext for Checker<'_> {
             }
         )
     }
+
+    fn in_loop_context(&self) -> bool {
+        let statements = self.current_statements();
```

Comment:
I don't think we need to track a separate `Vec` of current statements on the `Checker`. This should already be available on the `SemanticModel` field, as in the code I linked earlier.


```suggestion
        let statements = self.semantic.current_statements();
```

My earlier suggestion about adding a `current_statements` method was not for `Checker`, I meant that instead of `SemanticSyntaxContext::in_loop_context`, we could consider adding `SemanticSyntaxContext::current_statements`, which would return an iterator over `Stmt` nodes:

```rust
fn current_statements(&self) -> impl Iterator<Item = &Stmt> {
    self.semantic.current_statements()
}
```

This should be easier to implement for both Ruff and ty, and then the actual `in_loop_context` check could be shared between the two. But I think `SemanticSyntaxContext::in_loop_context` is also okay, especially if you tried this already.